### PR TITLE
Remove obsolescent and non-standard features in physics modules

### LINKED ISSUE #1270: Obsolescent and non-standard features in several physics modules
**Describe the bug**

I'm porting v4.2.1 to IBM Power9 with the XL compiler suite. During the process, I uncovered and fixed several obsolescent and non-standard features used in physics modules, specifically:

* phys/module_cu_mskf.F: `amax1` intrinsic (obsolescent in F2018) used and is given arguments of the wrong type.
* phys/module_ra_goddard.F: Arguments of the wrong type passed to `min` and `max` intrinsics.
* phys/module_mp_fast_sbm.F:  `amax1` intrinsic (obsolescent in F2018) used and is given arguments of the wrong type.
* phys/module_dust_emis.F: Arguments of the wrong type passed to `min` and `max` intrinsics.
* phys/module_cu_scalesas.F: Arguments of the wrong type passed to `min` intrinsic.
* phys/module_mp_p3.F: Uses non-standard subroutine `flush`.
* phys/module_shcu_deng.F: Uses non-standard subroutine `flush`.

The fixes were straightforward:

* replacing `amax1` with `max`;
* explicitly casting literal constants to the real kind used by the other argument in `min` or `max` intrinsics;
* replacing `call flush()` with the `flush()` statement.

**Would you like a PR?**

**To Reproduce**

Compile using IBM XL v16.1.1 on Power9.

Here's a small sample of the error message:

```
"module_cu_mskf.f90", line 1765.26: 1513-041 (S) Arguments of the wrong type were specified for the INTRINSIC procedure "amax1".
"module_cu_mskf.f90", line 1766.27: 1513-041 (S) Arguments of the wrong type were specified for the INTRINSIC procedure "amax1".
** module_cu_mp   === End of Compilation 1 ===
```
and similar for other modules.

Errors related to non-standard subroutine `flush` appear at the linking stage.

**Expected behavior**

WRF should compile. It compiles and runs after fixes.

**Screenshots**
n/a

**Attachments**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by milancurcic**
TYPE: Bug fix

KEYWORDS: Fix, obsolescent, non-standard, features

SOURCE: Milan Curcic (University of Miami)

DESCRIPTION OF CHANGES:
Problem:
This PR removes or fixes some obsolescent and non-standard features in several physics modules which were 
preventing WRF to be built with IBM XL v16.1.1 on Power9, as reported in #1270.

Solution:
* Replace `amax1()` occurrences with `max()`
* Explicitly cast literal constants that are passed to `min()` and `max()` intrinsics so that they are type and kind 
compatible with other arguments
* Replace calls to the non-standard `flush()` subroutine with the standard `flush()` statement.

ISSUE:
Fixes #1270 

LIST OF MODIFIED FILES: 
M       phys/module_cu_mskf.F
M       phys/module_cu_scalesas.F
M       phys/module_dust_emis.F
M       phys/module_mp_fast_sbm.F
M       phys/module_mp_milbrandt2mom.F
M       phys/module_mp_p3.F
M       phys/module_mp_thompson.F
M       phys/module_ra_goddard.F
M       phys/module_shcu_deng.F

TESTS CONDUCTED: 
1. These fixes were necessary to allow WRF to be built and run on IBM Power9 with the XL v16.1.1 compiler. The 
executables build in both single (`./configure`) and double (`./configure -r8`) precision modes.
2. Jenkins testing is all pass.

RELEASE NOTE: Several physics modules were updated to be more Fortran standard-conforming. This PR removes or fixes some obsolescent and non-standard features in several physics modules which were preventing WRF to be built with IBM XL v16.1.1 on Power9.

### CODE REVIEW

**Comment by dudhia** [comment]
Dave will need to confirm, but it looks fine to me and probably can be targeted for release 4.2.2.

**Comment by milancurcic** [comment]
Oops, I only now see that I based this off master. We can rebase it to release-v4.2.2 if needed.

**Comment by dudhia** [comment]
Should not be much difference. master is probably 4.2.1

**Comment by dudhia** [comment]
should be sufficient to just change the target in the PR.

**Comment by milancurcic** [comment]
@dudhia @weiwangncar @davegill Does this PR need any more care or is it good to go?

**Comment by davegill** [comment]
@milancurcic 
> Does this PR need any more care or is it good to go?

Milan,
Do your mods work when building with double precision? Since our compilers do not catch this, we rely on you. 
```
> ./clean -a
> ./configure -r8
> ./compile em_real >& foo
```

**Comment by milancurcic** [comment]
@davegill Good call. `-r8` triggers similar errors in other physics modules. Stand by for more fixes.

**Comment by milancurcic** [comment]
Both single and double precision modes now build. I updated the PR description with new files that are changed.

**Comment by weiwangncar** [comment]
I'm ok with this PR.

**Comment by weiwangncar** [comment]
@davegill @dudhia Any more comments on this PR? Shall I approve and merge it?

**Comment by dudhia** [comment]
I wonder if it should go to develop instead of 4.2.2

**Comment by weiwangncar** [comment]
@dudhia @davegill I can agree to that, making this PR to develop branch.

**Comment by davegill** [comment]
@weiwangncar @dudhia 
Folks,
I think that this should be directed at the release-v4.2.2 branch, not develop. This is definitely a bug fix. I am OK with this, but I do not want to approve if you guys are still deciding which branch to use.

I removed the un-associated mod to the .gitignore file. This is now completely a fix for compiling on power9 with IBM's xl compiler.

**Review by weiwangncar — APPROVED** [review]

**Comment by weiwangncar** [comment]
@davegill I have approved it. So I'm ok for this to go to 4.2.2.

**Comment by dudhia** [comment]
I just saw a bunch of other developers who might need to be notified for
various physics options affected.

**Comment by davegill** [comment]
@dudhia 

> I just saw a bunch of other developers who might need to be notified for various physics options affected.

Is this a request to not merge the code?

**Comment by dudhia** [comment]
This is why I wasn’t sure

**Comment by davegill** [comment]
@dudhia @weiwangncar 
Jimy,
Understood.

I think that since these are bug fixes, we notify the developers to review this PR. There are some schemes that we can say that "we" own.

**Comment by JS-WRF-SBM** [comment]
@davegill @milancurcic @dudhia @weiwangncar 

Folks, thanks for working on these corrections.
I'm OK with the changes for the `phys/module_mp_fast_sbm.F` scheme in this PR.

Koby

**Comment by llpcarson** [comment]
Looks ok to me (module_cu_scalesas.F )

**Comment by davegill** [comment]
@dudhia @weiwangncar 
Folks,
We have heard back from almost all impacted. In addition to those above, Hugh, Janson, and Kiran sent "OK" emails. I say we go for it with these mods.

**Comment by weiwangncar** [comment]
I'm ok with this, and have already approved this PR.

**Comment by davegill** [comment]
@dudhia @weiwangncar 
Jimy,
Wei has already approved this. Most of the impacted physics developers have given a thumbs up. This is OK to merge

**Comment by dudhia** [comment]
OK

**Pull request merged**
